mmv1/third_party/terraform/services/composer/data_source_google_composer_user_workloads_secret_test.go.erb

```diff
@@ -0,0 +1,100 @@
+<% autogen_exception -%>
+package composer_test
+
+<% unless version == 'ga' -%>
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDataSourceComposerUserWorkloadsSecret_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"env_name":    fmt.Sprintf("%s-%d", testComposerEnvironmentPrefix, acctest.RandInt(t)),
+		"secret_name": fmt.Sprintf("%s-%d", testComposerUserWorkloadsSecretPrefix, acctest.RandInt(t)),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceComposerUserWorkloadsSecret_basic(context),
+				Check: resource.ComposeTestCheckFunc(
+					checkSecretDataSourceMatchesResource(),
+				),
+			},
+		},
+	})
+}
+
+func checkSecretDataSourceMatchesResource() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ds, ok := s.RootModule().Resources["data.google_composer_user_workloads_secret.test"]
+		if !ok {
+			return fmt.Errorf("can't find %s in state", "data.google_composer_user_workloads_secret.test")
+		}
+		rs, ok := s.RootModule().Resources["google_composer_user_workloads_secret.test"]
+		if !ok {
+			return fmt.Errorf("can't find %s in state", "google_composer_user_workloads_secret.test")
+		}
+
+		dsAttr := ds.Primary.Attributes
+		rsAttr := rs.Primary.Attributes
+		errMsg := ""
+
+		for k := range rsAttr {
+			if k == "%" {
+				continue
+			}
+			// ignore diff if it's due to secrets being masked.
```

Comment:
why would the secrets be masked?

Reply:
The API is returning masked values in "data" field. But they are still stored in state in case it was set in Terraform configuration.